src/common/common.ts

```diff
@@ -396,6 +405,394 @@ export async function getVerifiableCredentialApiConfiguration(
   };
 }
 
+/**
+ * @param response Takes a response from a VC service and checks that it has the correct status and content type
+ * @param vcUrl The URL of the VC
+ * @returns The input response
+ */
+function validateVcResponse(response: Response, vcUrl: string): Response {
+  if (!response.ok) {
+    throw new Error(
+      `Fetching the Verifiable Credential [${vcUrl}] failed: ${response.status} ${response.statusText}`,
+    );
+  }
+
+  const contentType = response.headers.get("Content-Type");
+  if (!contentType) {
+    throw new Error(
+      `Fetching the Verifiable Credential [${vcUrl}] failed: Response does not have a Content-Type header; expected application/ld+json`,
+    );
+  }
+
+  const parsedContentType = contentTypeParser.parse(contentType);
+  const [mediaType, subtypesString] = parsedContentType.type.split("/");
+  const subtypes = subtypesString.split("+");
+
+  if (mediaType !== "application" || !subtypes.includes("json")) {
+    throw new Error(
+      `Fetching the Verifiable Credential [${vcUrl}] failed: Response has an unsupported Content-Type [${contentType}]; expected application/ld+json`,
```

Comment:
```suggestion
      `Fetching the Verifiable Credential [${vcUrl}] failed: Response has an unsupported Content-Type [${contentType}]; expected application/json`,
```

Currently we are only enforcing `application/json`. Note that the `ld` part _can_ be enforced once the [V2.0 VC Data Model](https://www.w3.org/TR/vc-data-model-2.0/) is used.